File: Java proyects/Positivo.java
import javax.swing.JOptionPane;
public class Positivo {
    public static void main(String[] args) {
        JOptionPane.showMessageDialog(null, "Algoritmo que imprime solo datos positivos");
        float[] n1 = new float[20];
        float[] p = new float[20];
        int pos = 0, rep;
        for (rep = 0; rep < 20; rep++) {
            n1[rep] = Float.parseFloat(JOptionPane.showInputDialog("Diga el dato #" + (rep + 1)));
            if(n1[rep]>0){
                p[pos]=n1[rep];
                pos=pos+1;
            }
        }
        if(pos>0){
            for(rep=0;rep<pos;rep++){
                JOptionPane.showMessageDialog(null,"Casilla "+ rep+ " Número "+ p[rep] );
            }
        }
        else{
            JOptionPane.showMessageDialog(null,"Todos los números eran negativos");
        }
    }
}

File: Phyton proyects/Positivo.py
print("Algoritmo que imprime solo datos positivos")
n1 = [0 for a in range(20)]
p = [0 for b in range(20)]
pos = 0
for rep in range(20):
    n1[rep] = float(input("Diga el dato #" + str(rep + 1)))
    if n1[rep] > 0:
        p[pos] = n1[rep]
        pos = pos + 1
if pos > 0:
    for c in range(pos):
        print("Casilla ", c, " Número ", p[c])
else:
    print("Todos los números eran negativos")
